sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,330 @@
+
+                prf = Profile(name=name, is_bot=is_bot)
+                uid.profile = prf
+
+                # now it is time to add the identities for name, emails and backends
+                sh_identities = __create_sh_identities(name, emails, yid)
+                uid.identities += sh_identities
+
+                if enrollments:
+                    affiliations = self.__parse_affiliations_yml(enrollments, uuid)
+                    uid.enrollments += affiliations
+
+                self._identities[uuid] = uid
+
+        except KeyError as e:
+            msg = "invalid GrimoireLab yaml format. Attribute %s not found" % e.args
+            raise InvalidFormatError(cause=msg)
+
+    def __parse_organizations(self, stream):
+        """Parse GrimoireLab organizations.
+
+        The GrimoireLab organizations format is a YAML element stored
+        under the "organizations" key. The next example shows the
+        structure of the document:
+
+        - organizations:
+            Bitergia:
+                - bitergia.com
+                - support.bitergia.com
+                - biterg.io
+            LibreSoft:
+                - libresoft.es
+
+        :param json: YAML object to parse
+
+        :raises InvalidFormatError: raised when the format of the YAML is
+            not valid.
+        """
+        if not stream:
+            return
+
+        yaml = self.__load_yml(stream)
+
+        try:
+            for element in yaml:
+                name = self.__encode(element['organization'])
+
+                if not name:
+                    msg = "invalid GrimoireLab yaml format. Empty organization name"
+                    raise InvalidFormatError(cause=msg)
+
+                o = Organization(name=name)
+
+                if 'domains' in element:
+                    if not isinstance(element['domains'], list):
+                        msg = "invalid GrimoireLab yaml format. List of elements expected for organization %s" % name
+                        raise InvalidFormatError(cause=msg)
+                    for dom in element['domains']:
+                        if dom:
+                            d = Domain(domain=dom, is_top_domain=False)
+                            o.domains.append(d)
+                        else:
+                            msg = "invalid GrimoireLab yaml format. Empty domain name for organization %s" % name
+                            raise InvalidFormatError(cause=msg)
+
+                self._organizations[name] = o
+        except KeyError as e:
+            msg = "invalid GrimoireLab yaml format. Attribute %s not found" % e.args
+            raise InvalidFormatError(cause=msg)
+
+        except TypeError as e:
+            msg = "invalid GrimoireLab yaml format. %s" % e.args
+            raise InvalidFormatError(cause=msg)
+
+    def __parse_affiliations_yml(self, affiliations, uuid):
+        """Parse identity's affiliations from a yaml dict."""
+        enrollments = []
```

Comment:
Leave a blank space in between the docstring and the next statement. When docstrings are just one line, we add this line.